pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+// this was copied from where else and edited to fit our objects
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+/* Begin Create Definition */
+
+type brokerCreateStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+	kapi.NameGenerator  // GenerateName method for CreateStrategy
+}
+
+// implements RESTCreateStrategy interface
+var createStrategy = brokerCreateStrategy{
+	// embeds to pull in existing code behavior from upstream
+
+	// this has an interesting NOTE on it. Not sure if it applies to us.
+	ObjectTyper: kapi.Scheme,
+	// use the generator from upstream k8s, or implement method
+	// `GenerateName(base string) string`
+	NameGenerator: kapi.SimpleNameGenerator,
+}
+
+// Canonicalize is called after validate. What happens if it creates
+// an object to persist that does not pass validate? (I think the
+// answer is "don't do that"). Frequently is an empty method or a type
+// check. May mutate the object.
+func (brokerCreateStrategy) Canonicalize(obj runtime.Object) {}
+
+// Are brokers namespace scoped?
+func (brokerCreateStrategy) NamespaceScoped() bool {
+	return false
+}
+
+// PrepareForCreate recieves
+func (brokerCreateStrategy) PrepareForCreate(ctx kapi.Context, obj runtime.Object) {
+	// coerce to our specific object type. (Should we type check?)
+	broker := obj.(*sc.Broker)
```

Comment:
Sure, one of the decisions I wanted us to make. None of the ones I see in k8s code are checked. Often the entire method body is an anonymous variable:
`_  := obj.(*sc.Broker)`
My understanding is this would cause a `panic` somewhere in go if it was violated, so it must never push a wrong object into this. I do not know where that happens.

Logging sounds good though.